src/devices/Qmc5883l/Qmc5883l.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.IO;
+using System.Numerics;
+using UnitsNet;
+
+namespace Iot.Device.Qmc5883l
+{
+    /// <summary>
+    /// 3-Axis Digital Compass QMC5883L
+    /// </summary>
+    [Interface("3-Axis Digital Compass QMC5883L")]
+    public class Qmc5883l : IDisposable
+    {
+        /// <summary>
+        /// QMC5883L Default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x0D;
+        private I2cDevice _i2cDevice;
+        private bool _isDeviceInitialized = false;
+
+        /// <summary>
+        /// Enables or disables the Interrupt PIN.
+        /// </summary>
+        /// <value>Disabled by default.</value>
+        public Interrupt Interrupt { get; set; } = Interrupt.Disable;
+
+        /// <summary>
+        /// When the point roll-over function is enabled, the I2C data pointer automatically rolls between 00H ~ 06H.
+        /// </summary>
+        /// <value>Disabled by default.</value>
+        public RollPointer RollPointer { get; set; } = RollPointer.Disable;
```

Comment:
this doesn't seem to be updating anything in the device, most of the devices are updating registers here. You are requiring people to call SetMode instead, most people won't remember to do it and will likely cause them to work differently than they wanted